### ISSUE TITLE
Real-time updating grows sluggish over time

### ISSUE THREAD

**Issue by tyrauber** [tradingview/lightweight-charts#338]
**Lightweight Charts Version:**  1.0.0

**Steps/code to reproduce:**

When updating a chart over websockets with live data using `series.update()`, the chart becomes sluggish overtime.

```javascript
areaSeries.update({
  time: date,
  value: newVal
});
```

I suspect, please correct me if I am wrong, that the data is pushed to an array that continues to grow overtime using up the browser memory, even though only the last X data points are rendered on the chart.

Another approach is to keep an array of data points and truncate it at an arbitrary length, but then the the chart data needs to be constantly overwritten with setData.

```
 seriesData.push({
   time: date,
  value: newVal,
});
if (seriesData.length > 30){
  seriesData.splice(0, 1)
}
series.setData(seriesData)
```

Is there a way to truncate data rendered off chart and still use update()?

**Comment by timocov** [comment]
Hi,

can you 1) try it with the latest version of the library (1.2.1 at the moment) 2) provide link to page where we can debug it (jsfiddle-like example is enough, or even any your page but please with un-minified/development code of the library) please?

**Comment by timocov** [comment]
Hey @tyrauber, do you have any updates here?

**Comment by tyrauber** [comment]
@timocov, It's hard to say.  Our App is rendering a half a dozen charts at once from a steady stream of data over web socket. We updated to 1.3.0 and also streamlined the cross communication between components, which did seem to improve performance. 

I might try creating a demonstration of 6 charts driven by real-time mock data, but until I got  something to share, it's probably appropriate to mark this one closed.

Thanks for following up!
